src/ott/problems/linear/barycenter_problem.py

```diff
@@ -69,29 +69,38 @@ def __init__(
     if self._is_segmented:
       # (num_measures, max_measure_size, ndim)
       # (num_measures, max_measure_size)
-      assert self._y.shape[:2] == self._b.shape
+      assert self._y.shape[:2] == self._b.shape, \
+        "Point clouds and weights do not have matching shapes"
     else:
       # (num_total_points, ndim)
       # (num_total_points,)
-      assert self._b is None or self._y.shape[0] == self._b.shape[0]
+      assert self._b is None or self._y.shape[0] == self._b.shape[
+          0], "Point clouds and weights do not have matching shapes"
 
   @property
-  def segmented_y_b(self) -> Tuple[jnp.ndarray, jnp.ndarray]:
+  def segmented_y_b(self) -> Tuple[jnp.ndarray, jnp.ndarray, jnp.ndarray]:
     """Tuple of arrays containing the segmented measures and weights.
 
     - Segmented measures of shape ``[num_measures, max_measure_size, ndim]``.
     - Segmented weights of shape ``[num_measures, max_measure_size]``.
     """
     if self._is_segmented:
-      y, b = self._y, self._b
+      y, b, num_per_measure = self._y, self._b, [self._y.shape[1]
```

Comment:
Please add `num_per_measure` on a separate line (+ use a jnp.array or modify the type).